### PR TITLE
♻️ Add middleware parameter

### PR DESCRIPTION
**Pull request by Kludex**

### CODE REVIEW

File: fastapi/routing.py [hunk 1]
```diff
@@ -795,6 +804,7 @@ def __init__(
                 """
             ),
         ] = Default(generate_unique_id),
+        middleware: Optional[Sequence[Middleware]] = None,
```
Comment (hasansezertasan) [inline fastapi/routing.py:807]: All other arguments have annotated documentation but this line misses it.

**Review by hasansezertasan — APPROVED** [review]
LGTM

**Comment by CCorrado** [comment]
@Kludex any chance you have a moment to update this PR? Nice job with it :+1:

**Comment by Kludex** [comment]
No. I've already let @tiangolo know about it.

**Comment by najork** [comment]
I'd love to see this make it into the next release, is that possible?

**Comment by aswenson86** [comment]
+1 would like to see this feature pulled into the next release as well.

**Comment by muazhari** [comment]
Any updates? I need it T_T.

**Comment by koogle** [comment]
@Kludex @hasansezertasan since I also would love to use this to simply a bunch of code we have I have fixed lint and added a basic documentation here https://github.com/tiangolo/fastapi/pull/11398

**Comment by hrazjan** [comment]
Is there any progress since last month? We could really use this in our project. It would make our code cleaner.

**Comment by koogle** [comment]
@tiangolo  closed my PR as I think they are refactoring that part of the code base, I think he ideally can give an update and a recommendation

**Comment by 10-mahmoud** [comment]
Bumping this. @Kludex would you mind getting the lint CI passing so @tiangolo can merge this ASAP? Thanks!

**Comment by Kludex** [comment]
https://github.com/tiangolo/fastapi/pull/11010#issuecomment-1919143645

**Comment by 10-mahmoud** [comment]
> [#11010 (comment)](https://github.com/tiangolo/fastapi/pull/11010#issuecomment-1919143645)

That's cool that you let @tiangolo know about it 6 months ago, but I think maybe 'tian might have got it lost since then? :stuck_out_tongue_winking_eye:

**Comment by darkpersoft** [comment]
Any news? It's very nice feature...

**Comment by Kludex** [comment]
> > [#11010 (comment)](https://github.com/tiangolo/fastapi/pull/11010#issuecomment-1919143645)
> 
> That's cool that you let @tiangolo know about it 6 months ago, but I think maybe 'tian might have got it lost since then? 😜

Blz. 👍

**Comment by Dorbmon** [comment]
Anything new about this? This feature is quite important. Thanks.

**Comment by Kludex** [comment]
> @Kludex: do you have time to update this one? Once it passes the tests we can put it back on Tiangolo's board 😇 🙏

No.

**Comment by svlandeg** [comment]
Ok, then I'll have a look!

**Comment by WilliamStam** [comment]
yikes. this has become a real necessity for us :( been able to side step it for the past year or so but no more. please please make it in :P you can do it little code. you can be big and strong!

**Comment by Kludex** [comment]
I've unsubscribed from this PR. If I'm required, ping my @.

**Pull request closed without merging**